numpy/lib/_index_tricks_impl.py

```diff
@@ -165,7 +165,7 @@ def __getitem__(self, key):
                     size.append(int(step))
                 else:
                     size.append(
-                        int(math.ceil((stop - start) / (step * 1.0))))
+                        math.ceil((stop - start) / (step * 1.0)))
```

Comment:
maybe I'm missing something, but the `* 1.0` looks like a Python 2 remnant to me

```suggestion
                        math.ceil((stop - start) / step))
```

Reply:
Same a few lines later:
https://github.com/numpy/numpy/blob/e151f0daf61120a7e3fcbd5ae5b7158dca355d60/numpy/lib/_index_tricks_impl.py#L186

Perhaps this could to be handled in a different PR. Right now I don't have a clear idea how to identify all cases of integer division that could be simplified.